src/snapred/backend/recipe/algorithm/CalibrationNormalizationAlgo.py

```diff
@@ -0,0 +1,169 @@
+import json
+
+from mantid.api import (
+    AlgorithmFactory,
+    ITableWorkspaceProperty,
+    MatrixWorkspaceProperty,
+    PropertyMode,
+    PythonAlgorithm,
+    mtd,
+)
+from mantid.kernel import Direction
+
+from snapred.backend.dao.ingredients.NormalizationCalibrationIngredients import (
+    NormalizationCalibrationIngredients as Ingredients,
+)
+from snapred.backend.recipe.algorithm.LiteDataCreationAlgo import LiteDataCreationAlgo
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+from snapred.backend.recipe.algorithm.RawVanadiumCorrectionAlgorithm import RawVanadiumCorrectionAlgorithm
+from snapred.backend.recipe.algorithm.SmoothDataExcludingPeaksAlgo import SmoothDataExcludingPeaks  # noqa F401
+
+name = "CalibrationNormalizationAlgo"
+
+
+class CalibrationNormalization(PythonAlgorithm):
+    def PyInit(self):
+        # declare properties
+        self.declareProperty(
+            MatrixWorkspaceProperty("InputWorkspace", "ws_in_normalization", Direction.Input, PropertyMode.Mandatory),
+            doc="Workspace containing the raw vanadium data",
+        )
+        self.declareProperty(
+            MatrixWorkspaceProperty("BackgroundWorkspace", "ws_background", Direction.Input, PropertyMode.Mandatory),
+            doc="Workspace containing the raw vanadium background data",
+        )
+        self.declareProperty(
+            MatrixWorkspaceProperty("OutputWorkspace", "ws_out_normalization", Direction.Output, PropertyMode.Optional),
+            doc="Workspace containing corrected data; if none given, the InputWorkspace will be overwritten",
+        )
+        self.declareProperty("Ingredients", defaultValue="", direction=Direction.Input)
+        self.declareProperty("FocusWorkspace", defaultValue="", direction=Direction.Output)
+        self.declareProperty("SmoothWorkspace", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def chopIngredients(self, ingredients: Ingredients):
+        self.reductionIngredients = ingredients.reductionIngredients
+        self.backgroundReductionIngredients = ingredients.backgroundReductionIngredients
+        self.calibrationRecord = ingredients.calibrationRecord
+        self.calibrantSample = ingredients.calibrantSample
+        self.calibrationWorkspace = ingredients.calibrationWorkspace
+        self.instrumentState = ingredients.instrumentState
+        self.focusGroup = ingredients.focusGroup
+        self.smoothDataIngredients = ingredients.smoothDataIngredients
+        yield
+
+    def PyExec(self):
+        ingredients = Ingredients(**json.loads(self.getProperty("Ingredients").value))
+        self.chopIngredients(ingredients)
+
+        runVanadiumFilePath = self.reductionIngredients.reductionState.stateConfig.vanadiumFilePath
+        backgroundVanadiumFilePath = self.backgroundReductionIngredients.reductionState.stateConfig.vanadiumFilePath
```

Comment:
these should probably be constructed the same way we would construct raw data paths in the `reductionAlgorithm`
e.g.
```
rawDataPath = ipts + "shared/lite/SNAP_{}.lite.nxs.h5".format(reductionIngredients.runConfig.runNumber)
```

Though I would prefer data be loaded outside the algo.

Reply:
It would probably be best to delegate this to the FetchGroceries chain.